src/overlays/actors/ovl_Dm_Char04/z_dm_char04.c

```diff
@@ -28,18 +30,125 @@ const ActorInit Dm_Char04_InitVars = {
     (ActorFunc)DmChar04_Draw,
 };
 
-#endif
+void func_80AABC40(SkelAnime* arg0, AnimationInfo* animation, u16 arg2) {
+    f32 phi_fv1;
+
+    animation += arg2;
+
+    if (animation->frameCount < 0.0f) {
+        phi_fv1 = Animation_GetLastFrame(animation->animation);
+    } else {
+        phi_fv1 = animation->frameCount;
+    }
+    Animation_Change(arg0, animation->animation, animation->playSpeed, animation->startFrame, phi_fv1, animation->mode,
+                     animation->morphFrames);
+}
+
+static AnimationInfo sAnimations[] = {
+    { &gameplay_keep_Anim_02B2E8, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, 0.0f },
+    { &gameplay_keep_Anim_029140, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, 0.0f },
+};
+
+Color_RGBAf sPrimColors[] = { { 250.0f, 255.0f, 230.0f, 255.0f },
+                              { 10.0f, 10.0f, 40.0f, 255.0f },
+                              { 255.0f, 235.0f, 220.0f, 255.0f } };
+Color_RGBAf sEnvColors[] = {
+    { 220.0f, 160.0f, 80.0f, 255.0f },
+    { 120.0f, 255.0f, 255.0f, 255.0f },
+    { 255.0f, 235.0f, 220.0f, 255.0f },
+};
+
+void DmChar04_Init(Actor* thisx, PlayState* play) {
+    DmChar04* this = THIS;
+
+    this->primColors = sPrimColors[this->actor.params];
+    this->envColors = sEnvColors[this->actor.params];
+    this->actor.targetArrowOffset = 3000.0f;
+    this->csAction = 0x63;
+    this->unk_262 = this->actor.params << 0xB;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    SkelAnime_Init(play, &this->skelAnime, (SkeletonHeader*)&gameplay_keep_Skel_02AF58, &gameplay_keep_Anim_029140,
+                   this->jointTable, this->morphTable, 7);
```

Comment:
```suggestion
    SkelAnime_Init(play, &this->skelAnime, &gameplay_keep_Skel_02AF58, &gameplay_keep_Anim_029140,
                   this->jointTable, this->morphTable, 7);
```

Reply:
It's necessary

Reply:
```suggestion
    SkelAnime_Init(play, &this->skelAnime, &gameplay_keep_Skel_02AF58.sh, &gameplay_keep_Anim_029140,
                   this->jointTable, this->morphTable, 7);
```
The first element of a `FlexSkeletonHeader` is the regular `SkeletonHeader` so you can just use that.